File: java/benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/RdmaReadThroughputOperation.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.Connection.Mode;
import de.hhu.bsinfo.observatory.benchmark.Connection.RdmaMode;

public class RdmaReadThroughputOperation extends RdmaThroughputOperation {

    RdmaReadThroughputOperation(Connection connection, Mode mode, int operationCount, int operationSize) {
        super(connection, mode, operationCount, operationSize, RdmaMode.READ);
    }

    @Override
    String getOutputFilename() {
        return "RDMA Read Throughput";
    }
}

File: cpp/src/observatory/operation/RdmaReadThroughputOperation.cpp
#include "RdmaReadThroughputOperation.h"

namespace Observatory {

RdmaReadThroughputOperation::RdmaReadThroughputOperation(Benchmark *benchmark, Benchmark::Mode mode, uint32_t operationCount, uint32_t operationSize) :
        RdmaThroughputOperation(benchmark, mode, operationCount, operationSize, Benchmark::RdmaMode::WRITE) {}

const char* RdmaReadThroughputOperation::getClassName() const {
    return "RdmaReadThroughputOperation";
}

const char* RdmaReadThroughputOperation::getOutputFilename() const {
    return "RDMA Read Throughput";
}

}